shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -923,7 +930,14 @@ - (void)setSelectedTextRangeLocal:(UITextRange*)selectedTextRange {
 
 - (void)setSelectedTextRange:(UITextRange*)selectedTextRange {
   [self setSelectedTextRangeLocal:selectedTextRange];
-  [self updateEditingState];
+
+  if (_enableDeltaModel) {
+    _previousTextEditingDelta = [_currentTextEditingDelta copy];
```

Comment:
Can you do `_previousTextEditingDelta = _currentTextEditingDelta` instead of copying?

Reply:
`FlutterTextEditingDelta` seems to be immutable (setDelta is exposed but doesn't seem to be called), so you can just pass its reference around without copying it I think?

Reply:
I tried that but since we flush `_currentTextEditingDelta` our `_previousTextEditingDelta` is also flushed.

Reply:
Oh this is cool! Thanks for the suggestion, I'm not too familiar with iOS but i'll try it.

Reply:
I looked into manual memory management, and tried again `_previousTextEditingDelta = _currentTextEditingDelta` and it is working as expected. I suspect my setting of `previousTextEditingDelta` in some other areas may have been caused this.